generic_config_updater/gu_common.py

```diff
@@ -212,6 +214,41 @@ def _invoke_validating_function(cmd, jsonpatch_element):
                 if not _invoke_validating_function(function, element):
                     raise IllegalPatchOperationError("Modification of {} table is illegal- validating function {} returned False".format(table, function))
 
+    def illegal_dataacl_check(self, old_config, upd_config):
+        # Block data ACL changes when patch includes:
+        # 1. table "type" being replaced
+        # 2. rule update on tables with table "type" replaced
+        # This will cause rase condition when swss consume the change
+        # and make acl rule inactive
+        old_acltable = old_config.get("ACL_TABLE", {})
+        upd_acltable = upd_config.get("ACL_TABLE", {})
+
+        # Pick data acl table with "type" field
+        old_dacltable = [table for table, fields in old_acltable.items()
+                         if fields.get("type") and fields["type"] != "CTRLPLANE"]
```

Comment:
>fields.get("type") [](http://example.com/codeflow?start=28&length=18)

This assumes that fields["type"] always exists if fields.get("type") is true. If not, suggest to use

if fields.get("type", None) and fields["type"] != "CTRLPLANE"]

Reply:
If not exists, fields.get("type") will return None. It is same as if fields.get("type", None)